### ISSUE TITLE
Assertion failed '!foundDiff'

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/runtime#38772]
Linux arm64

https://dev.azure.com/dnceng/public/_build/results?buildId=714263&view=ms.vss-test-web.build-test-results-tab&runId=22139762&resultId=174096&paneView=dotnet-dnceng.dnceng-build-release-tasks.helix-test-information-tab

```
COMPlus_TieredCompilation=0
COMPlus_JitStress=2
COMPlus_DbgMiniDumpName=/home/helixbot/dotnetbuild/dumps/coredump.%d.dmp
COMPlus_DbgEnableMiniDump=1
+ ./RunTests.sh --runtime-path /root/helix/work/correlation
----- start Thu Jul 2 08:08:53 UTC 2020 =============== To repro directly: =====================================================
pushd .
/root/helix/work/correlation/dotnet exec --runtimeconfig System.Diagnostics.TextWriterTraceListener.Tests.runtimeconfig.json --depsfile System.Diagnostics.TextWriterTraceListener.Tests.deps.json xunit.console.dll System.Diagnostics.TextWriterTraceListener.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing 
popd
===========================================================================================================
/root/helix/work/workitem /root/helix/work/workitem
  Discovering: System.Diagnostics.TextWriterTraceListener.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.Diagnostics.TextWriterTraceListener.Tests (found 60 test cases)
  Starting:    System.Diagnostics.TextWriterTraceListener.Tests (parallel test collections = on, max threads = 4)

Assert failure(PID 134 [0x00000086], Thread: 148 [0x0094]): Assertion failed '!foundDiff' in 'System.Diagnostics.TextWriterTraceListenerTests.DelimiterWriteMethodTestsBase:AttributeDelimiter_IgnoredAfterDelimiterSet_Test(System.String,System.Diagnostics.TraceFilter,System.Diagnostics.TraceEventCache,System.String,int,int,System.Object[]):this' during 'Linear scan register alloc' (IL size 146)

    File: /__w/1/s/src/coreclr/src/jit/lsra.cpp Line: 2263
    Image: /root/helix/work/correlation/dotnet
```

Happens in:
- System.Diagnostics.TextWriterTraceListener.Tests, JitStress=2, JitStress=1 + tiered
- System.Diagnostics.TraceSource.Tests, JitStress=2, Tiered

@CarolEidt

**Comment by safern** [comment]
Hit this again. @CarolEidt any update on this issue?

https://helix.dot.net/api/2019-06-17/jobs/802052ef-d341-4c6b-94fe-54771d721591/workitems/System.Net.Http.Functional.Tests/console

**Comment by CarolEidt** [comment]
@safern - I don't see the assert in the log that you linked to. What test run is this?

I've been unable to repro this on arm64, as per the original issue.
Note that the "!foundDiff" assert can be due to many things. One instance of that assert was reported in https://github.com/dotnet/runtime/issues/39221, and is due to a liveness issue with the "FramesRoot", but it's not clear that it's the same root cause.

**Comment by safern** [comment]
Ah, damnit my clipboard betrayed me 😢 -- this is the actual one: https://helix.dot.net/api/2019-06-17/jobs/f11a6a15-7cec-47ea-9592-33af900327ae/workitems/System.Net.Requests.Tests/console

Looks very similar to this one... so I think it is the same issue (maybe?)

**Comment by CarolEidt** [comment]
I believe that this was fixed with #39452.
